### PR TITLE
kubeadm: Introduce v1beta2 config

### PR DESCRIPTION
**Pull request by rosti**
**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:

This change adds v1beta2 config to kubeadm. Currently this is a 1:1 copy of v1beta1.

**Which issue(s) this PR fixes**:

Refs kubernetes/enhancements#970 kubernetes/kubeadm#1439

**Special notes for your reviewer**:

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/priority important-longterm
/assign @fabriziopandini
/assign @timothysc 
/assign @luxas 

**Does this PR introduce a user-facing change?**:

```release-note
Introduce the v1beta2 config format to kubeadm.
```

### CODE REVIEW

**Comment by rosti** [comment]
/hold
For kubernetes/enhancements#969 to merge.

**Comment by rosti** [comment]
/assign @fejta 
For the `hack/.golint_failures` change.

File: hack/.golint_failures [hunk 1]
```diff
@@ -3,6 +3,7 @@ cmd/kube-apiserver/app
 cmd/kube-controller-manager/app
 cmd/kube-proxy/app
 cmd/kubeadm/app/apis/kubeadm/v1beta1
+cmd/kubeadm/app/apis/kubeadm/v1beta2
```
Comment (fejta) [inline hack/.golint_failures]: This moves us in the wrong direction.

Please address the lint errors in your code and remove the exception on this package.

Comment (neolit123) [reply hack/.golint_failures]: my research on the problem here is the code generated by the k8s code generators do not pass the current state of go linters. they use underscores in generated function names such `SetDefaults_MyFunction`, which does not pass.
https://github.com/kubernetes/kubernetes/pull/76710/files#diff-9a2f5c3d2d4df5c049e0951f45d50133R69

this means that API namespaces need to be skipped.

Comment (rosti) [reply hack/.golint_failures]: @neolit123 is right here. I am not aware of any way of making the defaulter gen call funcs, that don't have underscores in their name and these obviously break the linter.

**Review by timothysc — APPROVED** [review]
/approve 

I'll let @neolit123 and @fabriziopandini go over with a fine-toothed comb.

File: hack/.golint_failures (hunk 1 above)
Comment (fejta) [reply hack/.golint_failures]: Is there an issue to fix this in the code generators?

Comment (neolit123) [reply hack/.golint_failures]: this PR seems to fix comment related lint issues, but not the `_`:
https://github.com/kubernetes/kubernetes/pull/59674

this ticket has a good enough title to be re-used:
https://github.com/kubernetes/code-generator/issues/30

i will re-open it and explain the `_` case.

**Comment by fejta** [comment]
/approve
/lgtm
thanks!

File: hack/.golint_failures (hunk 1 above)
Comment (fejta) [reply hack/.golint_failures]: Thanks!

File: cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+//	- ClusterConfiguration now has a "dns" field which can be used to select and configure the cluster DNS addon.
+//	- "featureGates" still exists under ClusterConfiguration, but there are no supported feature gates in 1.13.
+//	See the Kubernetes 1.13 changelog for further details.
+//	- Both "localEtcd" and "dns" configurations now support custom image repositories.
+//	- The "controlPlane*"-related fields in JoinConfiguration were refactored into a sub-structure.
+//	- "clusterName" was removed from JoinConfiguration and the name is now fetched from the existing cluster.
+//
+// Migration from old kubeadm config versions
+//
+// Please convert your v1alpha3 configuration files to v1beta2 using the "kubeadm config migrate" command of kubeadm v1.13.x
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go]: we'll need to have the doc.go as a TODO.
the search and replace is fine for now.

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Review by fabriziopandini — APPROVED** [review]
@rosti thanks for taking lead on this effort!
/approve
/lgtm

Only two nits on docs; up to you decide if to fix in this PR or in subsequent PRs

**Review by dixudx — APPROVED** [review]
/lgtm

**Comment by timothysc** [comment]
@rosti I'm ok with an `/hold cancel` when you are.

**Pull request merged**